src/api-reference/things/things.service.ts

```diff
@@ -68,6 +72,7 @@ export class ThingsService {
       this.eventsService.emitUpdated({ id, ...patch });
       return true;
     } else {
+      if (!dto.id || id !== dto.id) throw new MismatchIdException();
```

Comment:
```suggestion
      if ( id !== dto.id) throw new MismatchIdException();
```
I think you can escape the first check, right? unless we have a condition when both `id` and `dto.id` are `undefined `🤔 . But in that case, we should just verify that `id` must not be` undefined` I think it is more correct.

Reply:
Yup, I was thinking of those corner-cases when I added that check. However, I made several requests with different ids configurations and if both ids are `undefined`, the `requireValidThingDescription` throws the error. Hence, just removing the first validation is the way to go.

Reply:
> However, I made several requests with different ids configurations and if both ids are undefined, the requireValidThingDescription throws the error. Hence, just removing the first validation is the way to go.

Do we have these tests in the unit tests? 😺

Reply:
No! But as it is the API behavior being tested and not a single function, i believe those should be add in the e2e tests. I will added then and commit here later today.

Reply:
Done!